src/components/Layout/DrawerAppBar/DrawerAppBar.js

```diff
@@ -77,12 +123,34 @@ const StyledInputBase = styled(InputBase)(({ theme }) => ({
 
 function DrawerAppBar(props) {
     const navigate = useNavigate();
+    const dispatch = useDispatch();
     const [mobileOpen, setMobileOpen] = React.useState(false);
+    const {user} = useSelector(state => state.authReducer);
 
     const handleDrawerToggle = () => {
         setMobileOpen(!mobileOpen);
     };
 
+    const navItems = navigationItems.filter(item => item.role.includes(user?user.type:GUEST));
+
+    useEffect(() => {
+        const localStorageUser = JSON.parse(localStorage.getItem("user"));
+        const localStorageToken = localStorage.getItem("token");
+        if(!user && localStorageUser){
+            dispatch(loginSuccess({'user': localStorageUser, "token": localStorageToken}));
+        }
+    }, []);
```

Comment:
we shouldn't do any local storage manipulation in components like DrawerAppBar. It is just a UI component, this kind of logic should be in App or similar. In dataDo project we had a higher order component [(_)](https://reactjs.org/docs/higher-order-components.html) that loads important data, and while loading, displays a loading (splash) screen. But you can also do this just in App component, maybe have a `loaded = useState(false)` state while you get stuff from local storage (and maybe something else in the future), and just display a small loader while this data is loaded